Examples/BasicExample/Makefile

```diff
@@ -1,21 +1,36 @@
+# Declares the root of the repository the game source code lives in. Most
+# of the time, the Makefile should already be at the root, and this may
+# not be needed. However, this can be adjusted if the source code lives
+# elsewhere.
 REPO_ROOT := $(shell git rev-parse --show-toplevel)
-PRODUCT := build/BasicExample.pdx
-include $(REPO_ROOT)/Examples/swift.mk
+
+# Declares where PlaydateKit is stored. In this example, PDKIT_ROOT is the
+# same as REPO_ROOT. In some setups and projects, this may differ.
+PDKIT_ROOT := $(shell git rev-parse --show-toplevel)
+
+# Declares information about the product being build. PRODUCT_NAME will
+# likely be the name of the Swift package or game where the source code
+# lives. PRODUCT points to where the build Playdate package file (pdx)
+# will be stored.
+PRODUCT_NAME := BasicExample
+PRODUCT := build/$(PRODUCT_NAME).pdx
+
+include $(PDKIT_ROOT)/Examples/swift.mk
 
 # MARK: - Build PlaydateKit Swift Module
-build/Modules/playdatekit_device.o: $(REPO_ROOT)/Sources/PlaydateKit/*.swift
+build/Modules/playdatekit_device.o: $(PDKIT_ROOT)/Sources/PlaydateKit/*.swift
 	$(SWIFT_EXEC) $(SWIFT_FLAGS) $(SWIFT_FLAGS_DEVICE) -c $^ -emit-module -o $@
 
-build/Modules/playdatekit_simulator.o: $(REPO_ROOT)/Sources/PlaydateKit/*.swift
+build/Modules/playdatekit_simulator.o: $(PDKIT_ROOT)/Sources/PlaydateKit/*.swift
 	$(SWIFT_EXEC) $(SWIFT_FLAGS) $(SWIFT_FLAGS_SIMULATOR) -c $^ -emit-module -o $@
 
 # MARK: - Build BasicExample Swift Object
-build/basicexample_device.o: Sources/BasicExample/*.swift | build/Modules/playdatekit_device.o
+build/basicexample_device.o: Sources/$(PRODUCT_NAME)/*.swift | build/Modules/playdatekit_device.o
 	$(SWIFT_EXEC) $(SWIFT_FLAGS) $(SWIFT_FLAGS_DEVICE) -c $^ -o $@
 $(OBJDIR)/pdex.elf: build/basicexample_device.o
```

Comment:
How important are all these `.o` filenames? Should we introduce a variable (like `$(PRODUCT_NAME_LOWERCASE)`) for them too?

Reply:
I'm not entirely sure on how the .o files play into it, but I had been thinking about this as well. However, I'd like to see if there's a way we can have Make automatically make a lowercase version instead of us developers needing to do this. I'm not an expert on Make, so that might be challenging. Perhaps this can be tackled in a different PR once we have the proper knowledge?

Reply:
The name of the object file doesn't matter as long as you pass it to the linker for the appropriate build

Reply:
Worth noting, it doesn't so much matter what they are, but the cache folder is shared (because of the "parent" make file), so if you do copy the BasicExample folder, and don't change these names, and you've built and run the BasicExample, you'll get some funky warnings. I had to clean my build folder and change these names before my the warnings went away.